### PR TITLE
Changed test data generation for SNP data.

### PR DESCRIPTION
**Pull request by gijskant**
Vary the value of minorAllele in SnpDataByProbeCoreDb. This
would detect a previously undetected bug in SnpLzRow
(fixed in d178a3e6d0056492cf94cb1e9ed66dcf472a8656).

### CODE REVIEW

File: transmart-core-db-tests/src/groovy/org/transmartproject/db/dataquery/highdim/snp_lz/SnpLzTestData.groovy [hunk 1]
```diff
@@ -208,6 +208,16 @@ class SnpLzTestData {
                         it.chromosome == annotation.chromosome }
             assert snpInfo != null
 
+            // Choose value of minorAllele based on the allele count, to vary its
+            // value: to generate test cases for both 'A1' and 'A2'.
+            long a1Count = 0
+            long a2Count = 0
+            gtss.each {
+                a1Count += it.findAll( a1 ).size()
+                a2Count += it.findAll( a2 ).size()
```
Comment (cataphract) [inline transmart-core-db-tests/src/groovy/org/transmartproject/db/dataquery/highdim/snp_lz/SnpLzTestData.groovy]: I'm not sure what's happening here. Up we see that gtss has the format `['T T', 'T A', ...]`. Then `it` inside the closure must be something like `'T T'`. OTOH, `a1` is a String. So you get something like `'T T'.findAll { 'A' }.size()`, which always returns the number of characters in the first string, that is 3.

This function is also inconsistent: you left `def a1 = annotation.alt` at the top, but then `a1` may not be the alternative. It seems to me that the choice about which allele will be minor one should be made before `a1` and `a2` are assigned.

Comment (gijskant) [inline transmart-core-db-tests/src/groovy/org/transmartproject/db/dataquery/highdim/snp_lz/SnpLzTestData.groovy]: Okay, I'll have another look.
But as far as I understand, minor alleles are not always alternate alleles.

Comment (cataphract) [inline transmart-core-db-tests/src/groovy/org/transmartproject/db/dataquery/highdim/snp_lz/SnpLzTestData.groovy]: I never said that was the case; on the contrary, the comments on lines 201 and 202 concede as much.

**Comment by cataphract** [comment]
Incorporated into #6 . Closing

**Pull request closed without merging**